### DISCUSSION TITLE
What functions can be parallelized?

### DISCUSSION THREAD

**Discussion by hrj21** [ImmuneDynamics/Spectre#124] [! News]
Hi all,

Thanks again for all your hard work on Spectre. 

My question is perhaps a little naïve: are (or can) any of Spectre's functions parallelized across multiple cores? I'm thinking in particular some of the more resource intensive things like batch alignment, computing t-SNE/UMAP embeddings, or extracting summary statistics from large experiments. If so, does this work out of the box or do I need to use something like {parallelMap}, and do you have any examples?

I have access to a computing cluster, and I'm just wondering whether I will be able to get a big performance boost if I run my pipelines there.

Thanks!
Hefin

**Comment by ghar1821** [comment]
I would say anything that involve a for loop can easily be parallelise. Anything else requires more thought, but certainly not impossible. For dimensionality reduction, I think FITsne is already parallelised. So you can use that instead of tsne or UMAP. 

We can probably look into parallelising some features in the future.
